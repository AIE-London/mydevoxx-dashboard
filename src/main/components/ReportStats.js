/**
 * Created by TSADLER on 03/04/2017.
 */
import { Row, Col } from "react-flexbox-grid";
import styled from "styled-components";
import React, { Component } from "react";

const StatMainHeader = styled.h1`
    text-align: left;
    font-size: 2.5em;
    color: #fff;
    font-family: helvetica;
`;

const StatHeader = styled.h2`
    text-align: left;
    font-size: 1.5em;
    color: #fff;
    font-family: helvetica;
`;

const StatValue = styled.p`
    text-align: center;
    font-size: 2em;
    color: #fff;
    font-family: helvetica;
`;
const Orange = styled.span`
    color: #ff9e19;
`;

class ReportStats extends Component {
  render() {
    return (
      <Row center="xs">
        <Col xs={10}>

          <Col xs={2}>
            <Row start="xs">
              <StatMainHeader>MyReport</StatMainHeader>
            </Row>
          </Col>

          <Row center="xs">
            <Col xs={10}>

              <StatHeader>
                I spent <Orange>{this.props.minutes} minutes</Orange>
              </StatHeader>

              <StatValue>
                Attending <Orange>{this.props.talks} talks</Orange>
              </StatValue>

              <StatHeader>Learning what's new in...</StatHeader>

              <StatValue>{this.props.learning} </StatValue>

              <StatHeader>Networking with</StatHeader>

              <StatValue>
                {" "}<Orange>{this.props.attendees}</Orange> Attendees
              </StatValue>

            </Col>
          </Row>
        </Col>

      </Row>
    );
  }
}

export default ReportStats;
