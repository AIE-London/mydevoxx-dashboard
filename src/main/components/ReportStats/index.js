/**
 * Created by TSADLER on 03/04/2017.
 */
import React from 'react';
import {Grid, Row, Col} from 'react-flexbox-grid';
import styled from 'styled-components';

const StatMainHeader = styled.h1`
    text-align: left;
    font-size: 2.5em;
    color: #fff;
    font-family: helvetica;
`;

const StatHeader = styled.h2`
    text-align: left;
    font-size: 1.5em;
    color: #fff;
    font-family: helvetica;
`;

const StatValue = styled.p`
    text-align: center;
    font-size: 2em;
    color: #fff;
    font-family: helvetica;
`;
const Orange = styled.span`
    color: #ff9e19;
`;

class reportStats extends React.Component {
    render() {

        return <Row center="xs">
            <Col xs={10}>


                <Col xs={2}>
                    <Row start="xs">
                        <StatMainHeader>MyReport</StatMainHeader>
                    </Row>
                </Col>

                <Row center="xs">
                    <Col xs={10}>

                        <StatHeader>I spent <Orange>450 minutes</Orange></StatHeader>

                        <StatValue>Attending <Orange>9 Talks</Orange></StatValue>

                        <StatHeader>Learning what's new in...</StatHeader>

                        <StatValue>JavaScript, Java, Polymer </StatValue>

                        <StatHeader>Networking with</StatHeader>

                        <StatValue> <Orange>300</Orange> Attendees</StatValue>

                    </Col>
                </Row>
            </Col>

        </Row>
    }
}

export default reportStats;