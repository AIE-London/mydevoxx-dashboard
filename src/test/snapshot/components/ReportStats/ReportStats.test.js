/**
 * Created by TSADLER on 04/04/2017.
 */
import React from "react";
import ReactDOM from "react-dom";
import ReportStats from "../../../../main/components/ReportStats";
import renderer from "react-test-renderer";

test("ReportStats component", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<ReportStats title="Welcome to Devoxx" />, div);
  });
});
test("ReportStats component with title snapshot", () => {
  const tree = renderer
    .create(
      <ReportStats
        minutes="455"
        talks="10"
        learning="JS, Java, Polymer"
        attendees="345"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
