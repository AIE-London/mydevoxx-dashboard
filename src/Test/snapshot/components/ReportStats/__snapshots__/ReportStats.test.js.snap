exports[`test reportStats component with title snapshot 1`] = `
<div
  className="row center-xs">
  <div
    className="col-xs-10">
    <div
      className="col-xs-2">
      <div
        className="row start-xs">
        <h1
          className="iscJSK">
          MyReport
        </h1>
      </div>
    </div>
    <div
      className="row center-xs">
      <div
        className="col-xs-10">
        <h2
          className="eUofwH">
          I spent 
          <span
            className="kiuWfQ">
            455
             minutes
          </span>
        </h2>
        <p
          className="jlawfx">
          Attending 
          <span
            className="kiuWfQ">
            10
             talks
          </span>
        </p>
        <h2
          className="eUofwH">
          Learning what\'s new in...
        </h2>
        <p
          className="jlawfx">
          JS, Java, Polymer
           
        </p>
        <h2
          className="eUofwH">
          Networking with
        </h2>
        <p
          className="jlawfx">
           
          <span
            className="kiuWfQ">
            345
          </span>
           Attendees
        </p>
      </div>
    </div>
  </div>
</div>
`;
